### PR TITLE
config: support online change lock manager config

### PR DESCRIPTION
**Pull request by NingLin-P**
###  What have you changed?

This pr is part of online change config.
This pr add support for online change lock manager config by adding new tasks to lock manger ( 
`Task::ChangeConfig` and `Task::ChangeTTL`).

###  What is the type of the changes?

- New feature (a change which adds functionality)

###  How is the PR tested?

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

File: cmd/src/server.rs [hunk 1]
```diff
@@ -272,12 +272,14 @@ impl TiKVServer {
...
         gc_worker: &GcWorker<RaftKv<ServerRaftStoreRouter>>,
     ) -> Arc<ServerConfig> {
+        let mut cfg_controller = ConfigController::new(self.config.clone());
         // Create CoprocessorHost.
         let mut coprocessor_host =
             CoprocessorHost::new(self.config.coprocessor.clone(), self.router.clone());
 
         let lock_mgr = if self.config.pessimistic_txn.enabled {
             let lock_mgr = LockManager::new();
+            cfg_controller.register("pessimistic_txn", Box::new(lock_mgr.config_manager()));
```
Comment (overvenus) [inline cmd/src/server.rs]: `pessimistic_txn` or `pessimistic-txn`?

File: src/server/lock_manager/config.rs [hunk 2]
```diff
@@ -30,9 +37,51 @@ impl Config {
...
+    }
+}
+
+impl ConfigManager for LockManagerConfigManager {
+    fn dispatch(&mut self, mut change: ConfigChange) -> Result<(), Box<dyn Error>> {
+        match (
+            change.remove("wait_for_lock_timeout").map(Into::into),
+            change.remove("wake_up_delay_duration").map(Into::into),
+        ) {
+            (timeout @ Some(_), delay) => {
```
Comment (overvenus) [inline src/server/lock_manager/config.rs]: What if `delay` is None?

File: src/server/lock_manager/deadlock.rs [hunk 3]
```diff
@@ -319,6 +329,9 @@ impl Display for Task {
             ),
             Task::DetectRpc { .. } => write!(f, "Detect Rpc"),
             Task::ChangeRole(role) => write!(f, "ChangeRole {{ role: {:?} }}", role),
+            Task::ChangeTTL(ttl) => write!(f, "ChangeTTL {{ ttl: {:?} }}", ttl),
```
Comment (overvenus) [inline src/server/lock_manager/deadlock.rs]:
```suggestion
            Task::ChangeTTL(ttl) => write!(f, "ChangeTTL { ttl: {:?} }", ttl),
```

File: src/server/lock_manager/deadlock.rs [hunk 4]
```diff
@@ -319,6 +329,9 @@ impl Display for Task {
             ),
             Task::DetectRpc { .. } => write!(f, "Detect Rpc"),
             Task::ChangeRole(role) => write!(f, "ChangeRole {{ role: {:?} }}", role),
```
Comment (overvenus) [inline src/server/lock_manager/deadlock.rs]:
```suggestion
            Task::ChangeRole(role) => write!(f, "ChangeRole { role: {:?} }", role),
```

File: cmd/src/server.rs (hunk 1 above)
Comment (NingLin-P) [reply cmd/src/server.rs]: `pessimistic_txn` same as the `TiKvConfig` field name.

File: src/server/lock_manager/config.rs (hunk 2 above)
Comment (NingLin-P) [reply src/server/lock_manager/config.rs]: `WaiterManager` need to listening both update of `timeout` and `delay`, even one of them are not change (represent as `None` here), we still need to dispatch the change to it, and `WaiterManager::handle_config_change` will take care of various situations.

File: src/server/lock_manager/deadlock.rs (hunk 3 above)
Comment (NingLin-P) [reply src/server/lock_manager/deadlock.rs]: we need `{{` here to print out  `{`, so `Task::ChangeTTL` print something like `ChangeTTL { ttl: 100ms }`.

File: src/server/lock_manager/deadlock.rs [hunk 5]
```diff
@@ -758,6 +780,12 @@ where
         debug!("handle change role"; "role" => ?role);
         self.change_role(role);
     }
+
+    fn handle_change_ttl(&mut self, ttl: Duration) {
+        let mut inner = self.inner.borrow_mut();
+        inner.detect_table.reset_ttl(ttl);
+        info!("Dead lock detector config updated"; "ttl" => ?ttl);
```
Comment (youjiali1995) [inline src/server/lock_manager/deadlock.rs]:
```suggestion
        info!("Deadlock detector config changed"; "ttl" => ?ttl);
```

File: src/server/lock_manager/waiter_manager.rs [hunk 6]
```diff
@@ -512,6 +534,20 @@ impl WaiterManager {
...
+
+    fn handle_config_change(&mut self, timeout: Option<u64>, delay: Option<u64>) {
+        if let Some(timeout) = timeout {
+            self.default_wait_for_lock_timeout = timeout;
+        }
+        if let Some(delay) = delay {
+            self.wake_up_delay_duration = delay;
+        }
+        info!(
+            "Waiter manager config change";
```
Comment (youjiali1995) [inline src/server/lock_manager/waiter_manager.rs]:
```suggestion
            "Waiter manager config changed";
```

File: src/server/lock_manager/waiter_manager.rs [hunk 7]
```diff
@@ -512,6 +534,20 @@ impl WaiterManager {
...
+            "default_wait_for_lock_timeout" => timeout,
+            "wake_up_delay_duration" => delay
```
Comment (youjiali1995) [inline src/server/lock_manager/waiter_manager.rs]: Should print current config, not the parameters.

File: src/server/lock_manager/config.rs [hunk 8]
```diff
@@ -43,4 +92,103 @@ mod tests {
...
+        assert_eq!(rollback.right(), Some(true));
+        validate_waiter(&waiter, move |timeout: u64, delay: u64| {
+            assert_eq!(timeout, DEFAULT_TIMEOUT);
+            assert_eq!(delay, 500);
+        });
+        validate_dead_lock(&deadlock, move |ttl: u64| {
+            // dead lock ttl should not change
+            assert_eq!(ttl, DEFAULT_TIMEOUT);
+        });
+
```
Comment (youjiali1995) [inline src/server/lock_manager/config.rs]: Suggest adding a test -- only update wait_for_lock_timeoout.

**Pull request merged**